lang/cs/Org.Apache.REEF.Wake/Avro/ProtocolSerializer.cs

```diff
@@ -163,13 +184,14 @@ public byte[] Write(object message, long sequence)
         /// <param name="data">Byte array containing a header message and message to be deserialized.</param>
         /// <param name="observer">An object which implements the IObserver<>
         /// interface for the message being deserialized.</param>
-        public void Read<T>(byte[] data, IObserver<IMessageInstance<T>> observer)
+        public void Read(byte[] data, object observer)
```

Comment:
Why is it just `object`? Don't we always pass a message observer as a second parameter?

Reply:
The protocol serializer is designed to be a general class that serializes and deserializes any protocol and so the message observers implement the IObserver<> interface for a number of messages whose type is only known internally after initialization. 

We could have an interface that implements IObserver<object> and require that message observers implement that interface. Then if a message is received which the message observer does not implement, we would call OnNext(object);

We could also subclass the protocol serializer and make the subclass constructor take an class which implements IObserver<> for all of the messages in the actual protocol.